### DISCUSSION TITLE
Sidebar auto-collapse not working again: regression or not?

### DISCUSSION THREAD

**Discussion by elringus** [vuejs/vitepress#2879] [General]
Initially, open sidebar categories were not automatically collapsing when navigating to a page in other category. Then (around rc4) it started working. And now (rc10) it's broken again. I wasn't able to find an option which controls that, so I wonder if that's a regression or by design?

To be clear, I expect only the category of the currently open page to be unfolded and other automatically collapse after navigation (either via search or by clicking a link).

https://github.com/vuejs/vitepress/assets/2056864/27b41c52-0262-4290-9604-625752e2918a

**Comment by brc-dd** [comment]
It's by design. Auto-collapse is not intended. Groups currently auto-expand only. rc-4 had some regressions. The current behavior aligns with what we had since alpha and beta versions.

The reasoning was mainly because a user might want a particular group expanded and then they move out of that group and if that group collapses it might not be what the user expects. (basically there was no way for the user to keep a section open)

**Reply by elringus** [reply]
Thanks for clarification! Any chance the rc4 regression could be brought back as an option? Consider a category with lots of pages inside; when you navigate to a page inside other category, it remains open and user can't even see the category they're inside anymore. For example, open this page: https://naninovel.github.io/guide/configuration and click "custom configuration" link under the video.

**Reply by brc-dd** [reply]
How about we move it to screen? Like if you open a link, make the sidebar scroll and bring that section into view? Would that work for you? 👀

**Reply by elringus** [reply]
Personally, I prefer sidebar collapsing in this case (+ scrolling in case current category is still overflowing viewport would be a bonus), but that's just because I've got used to vuepress v1, which was doing that. Not sure what's better for most user's UX, to be honest. But at least scrolling would be nice to have.

**Reply by Karlstens** [reply]
I too suspect that in my lengthy sidebar, that auto-collapse would be desirable for the user whilst browsing outside of a previously expanded category. I would like to see it added as an option to https://vitepress.dev/reference/default-theme-sidebar
